diff --git a/include/ankerl/unordered_dense.h b/include/ankerl/unordered_dense.h
index e2d3851..361cd8c 100644
--- a/include/ankerl/unordered_dense.h
+++ b/include/ankerl/unordered_dense.h
@@ -885,7 +885,7 @@ private:
 
     [[nodiscard]] constexpr auto calc_shifts_for_size(size_t s) const -> uint8_t {
         auto shifts = initial_shifts;
-        while (shifts > 0 && static_cast<size_t>(static_cast<float>(calc_num_buckets(shifts)) * max_load_factor()) < s) {
+        while (shifts > 0 && static_cast<size_t>((static_cast<uint64_t>(calc_num_buckets(shifts)) * 4) / 5) < s) {
             --shifts;
         }
         return shifts;
@@ -925,7 +925,7 @@ private:
             // reached the maximum, make sure we can use each bucket
             m_max_bucket_capacity = max_bucket_count();
         } else {
-            m_max_bucket_capacity = static_cast<value_idx_type>(static_cast<float>(m_num_buckets) * max_load_factor());
+            m_max_bucket_capacity = static_cast<value_idx_type>((static_cast<uint64_t>(m_num_buckets) * 4) / 5);
         }
     }
 
